### PR TITLE
fix: do not expose internal errors to the client

### PR DESCRIPTION
**Pull request by soulbird**
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/wolf-rbac.lua [hunk 1]
```diff
@@ -274,7 +273,7 @@ function _M.rewrite(conf, ctx)
     core.log.info("token info: ", core.json.delay_encode(tokenInfo),
                   ", err: ", err)
     if err then
-        return 401, fail_response('invalid rbac token: parse failed')
+        return 401, fail_response('Invalid rbac token: parse failed')
```
Comment (spacewander) [inline apisix/plugins/wolf-rbac.lua:276]: Better avoid unnecessary change

File: apisix/plugins/ldap-auth.lua [hunk 2]
```diff
@@ -131,7 +131,8 @@ function _M.rewrite(conf, ctx)
 
     local user, err = extract_auth_header(auth_header)
     if err then
-        return 401, { message = err }
+        core.log.warn(err)
+        return 401, { message = "Invalid user authorization" }
```
Comment (spacewander) [inline apisix/plugins/ldap-auth.lua:135]:
```suggestion
        return 401, { message = "Invalid authorization in request" }
```
would be more match the "Missing authorization in request"?

File: apisix/plugins/basic-auth.lua [hunk 3]
```diff
@@ -151,13 +151,15 @@ function _M.rewrite(conf, ctx)
 
     local username, password, err = extract_auth_header(auth_header)
     if err then
-        return 401, { message = err }
+        core.log.warn(err)
+        return 401, { message = "Invalid user authorization" }
```
Comment (spacewander) [inline apisix/plugins/basic-auth.lua:155]: Ditto

File: apisix/plugins/wolf-rbac.lua [hunk 4]
```diff
@@ -326,7 +326,7 @@ function _M.rewrite(conf, ctx)
         core.log.error(" check_url_permission(",
             core.json.delay_encode(perm_item),
             ") failed, res: ",core.json.delay_encode(res))
-        return 401, fail_response(res.err,
+        return 401, fail_response("Invalid user authorization",
```
Comment (spacewander) [inline apisix/plugins/wolf-rbac.lua:329]:
```suggestion
        return 401, fail_response("Invalid user permission",
```

File: apisix/plugins/wolf-rbac.lua [hunk 5]
```diff
@@ -284,7 +283,8 @@ function _M.rewrite(conf, ctx)
 
     local consumer_conf = consumer.plugin(plugin_name)
     if not consumer_conf then
-        return 401, fail_response("Missing related consumer")
+        core.log.warn("Missing related consumer")
+        return 401, fail_response("Invalid user authorization")
```
Comment (spacewander) [inline apisix/plugins/wolf-rbac.lua:287]: the error here is nothing about the Authorization header

Comment (soulbird) [reply apisix/plugins/wolf-rbac.lua]: I'll reset this modification

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]
